test/span.js

```diff
@@ -182,6 +186,121 @@ describe('span should', () => {
+        ];
+        _.each(options, (o) => {
+            it (o.desc + 'should save tags, and logs on an unsampled span incase it later becomes sampled', () => {
+                let reporter = new InMemoryReporter();
+                let tracer = new Tracer(
+                    'test-service-name',
+                    reporter,
+                    new ConstSampler(false),
+                    { logger: new MockLogger() }
+                );
+                let span = tracer.startSpan('initially-unsampled-span');
+                span.setTag('tagKeyOne', 'tagValueOne');
+                span.addTags({
+                    'tagKeyTwo': 'tagValueTwo'
+                });
+                span.log({'logkeyOne': 'logValueOne'});
+
+                tracer._sampler = new ConstSampler(o.sampling);
+                span.setOperationName('sampled-span');
+                span.finish();
+
+                assert.isOk(deepEqual(span._tags[0], {key: 'tagKeyOne', value: 'tagValueOne'}));
+                assert.isOk(deepEqual(span._tags[1], {key: 'tagKeyTwo', value: 'tagValueTwo'}));
+                assert.isOk(deepEqual(span._logs[0].fields[0], {key: 'logkeyOne', value: 'logValueOne'}));
+                assert.equal(reporter.spans.length, o.reportedSpans);
+            });
+        });
+
+        describe('Finalize spans that', () => {
+            it ('inherit sampling decision is finalized', () => {
+                let childSpan = tracer.startSpan('child-span', {childOf: span});
+                assert.isOk(span.context().samplingFinalized);
+                assert.isOk(childSpan.context().samplingFinalized);
+            });
+
+            it ('set the sampling priority to enable debug', () => {
+                span.setTag(opentracing.Tags.SAMPLING_PRIORITY, 1);
+                assert.isOk(span.context().samplingFinalized);
+            });
+
+            it ('has finished', () => {
+                span.finish();
+                assert.isOk(span.context().samplingFinalized);
+            });
+
+            it ('has called setOperationName', () => {
+                span.setOperationName('fry');
+                assert.isOk(span.context().samplingFinalized);
+            });
+
+            it ('are injected into headers', () => {
+                let headers = {};
+                tracer.inject(span.context(), opentracing.FORMAT_HTTP_HEADERS, headers);
+
+                assert.isOk(span.context().samplingFinalized);
+            });
+        });
+
+        it ('isWriteable returns true if not finalized, or the span is sampled', () => {
+            tracer = new Tracer(
+                'test-service-name',
+                new InMemoryReporter(),
+                new ConstSampler(false),
+                { logger: new MockLogger() }
+            );
+            let unFinalizedSpan = tracer.startSpan('unFinalizedSpan');
+
+            assert.isOk(unFinalizedSpan._isWriteable());
+            assert.isOk(span._isWriteable());
```

Comment:
hard to follow - at least add a comment explaining that setup creates `span` as sampled, which is what you're relying on here.